nomad/structs/config/consul.go

```diff
@@ -359,6 +401,74 @@ func (c *ConsulConfig) Copy() *ConsulConfig {
 	if nc.AllowUnauthenticated != nil {
 		nc.AllowUnauthenticated = pointer.Of(*nc.AllowUnauthenticated)
 	}
+	if nc.UseIdentity != nil {
+		nc.UseIdentity = pointer.Of(*nc.UseIdentity)
+	}
 
 	return nc
 }
+
+// WorkloadIdentity is the jobspec block which determines if and how a workload
+// identity is exposed to tasks similar to the Vault block.
+//
+// This is a copy of WorkloadIdentity from nomad/structs package in order to
+// avoid import cycles.
```

Comment:
It may be good to add a similar comment in `structs` as a reminder to apply changes here as well.